lib/versioncontrol/github/github.go

```diff
@@ -188,35 +190,85 @@ func (i *Iterator) Next() bool {
 		// check unfiltered page size for halt condition
 		i.done = true
 	}
-	i.page = make([]Release, 0, len(page))
+	i.page = make([]vc.Target, 0, len(page))
 	for _, r := range page {
 		if !semver.IsValid(r.TagName) {
 			log.Debugf("Skipping non-semver release tag: %q\n", r.TagName)
 			continue
 		}
-		i.page = append(i.page, Release{
-			Version: r.TagName,
-		})
+		labels := parseReleaseNoteLabels(r.Body)
+		labels[vc.LabelVersion] = r.TagName
+		i.page = append(i.page, labels)
 
 		// only match `<major>.<minor>` when finding halt point. theoretically
 		// unnecessary, but this feels a bit less brittle than halting on a specific
 		// tag, and is more consistent than using a gt/lt rule, since we occasionally
 		// release patches for very old versions.
 		if semver.MajorMinor(r.TagName) == semver.MajorMinor(i.halt) {
+			// set 'done' so that this is the last page we end up processing
 			i.done = true
-			break
 		}
 	}
 	return i.err == nil && len(page) != 0
 }
 
 // Page loads the current page. Must not be called until after Next() has been
 // called. Subsequent calls between calls to Next() return the same value.
-func (i *Iterator) Page() []Release {
+func (i *Iterator) Page() []vc.Target {
 	return i.page
 }
 
 // Error checks if an error occurred during iteration.
 func (i *Iterator) Error() error {
 	return i.err
 }
+
+// parseReleaseNoteLabels attempts to extract labels from github release notes.
+// Invalid values are skipped/ignored in order to ensure that future extensions of
+// the label format can be resonably backwards-compatible.
+func parseReleaseNoteLabels(notes string) map[string]string {
+
+	const labelPrefix = "labels:"
+	lines := strings.FieldsFunc(notes, func(r rune) bool {
+		return r == '\r' || r == '\n'
+	})
+
+	labels := make(map[string]string)
+
+	for _, l := range lines {
+		l = strings.ToLower(strings.TrimSpace(l))
+		if !strings.HasPrefix(l, labelPrefix) {
+			continue
+		}
+		l = strings.TrimPrefix(l, labelPrefix)
+		for _, kv := range strings.Split(l, ",") {
```

Comment:
I considered it.  Opted not to because I didn't want changes in CLI syntax to have crossover with this functionality.  Seemed like minimal extra effort to keep them distinct.